wordpress-mobile/WordPress-iOS
# Refactor `NotificationsViewController` & Add `NotificationsViewModel`

Refs: pcdRpT-5aB-p2

This PR is a first of a following chain to reduce the complexity of `NotificationsViewController` by attempting to decouple logic from the view. So we can add and modify features quickly in our next project. 

I will isolate the changes semantically to make testing and reviewing easier. So all the changes here are about `lastSeen` logic.

## Testing Steps
1. Install & Login to Jetpack app.
2. App badges must be enabled. Check current badge count.
3. Navigate to Notifications
4. ✅ Check if viewing the Notifications updates the badges as it should.

## Regression Notes
1. Potential unintended areas of impact
The refactored areas themselves

5. What I did to test those areas of impact (or what existing automated tests I relied on)
Unit tests added.

6. What automated tests I added (or what prevented me from doing so)

PR submission checklist:
